docs/content/docs/build/data-modeling.md

```diff
@@ -3,26 +3,107 @@ title: Data modeling
 
-## Patterns for modeling data
+### Using handles to store and retrieve Fluid objects
 
-### When to use initialObjects vs dynamicObjects
+All shared objects have a `handle` property that can be used to store and retrieve them from other shared objects. Objects created dynamically must be stored before they are collaborative. As you will see below, the act of storing a handle is what links the new dynamic object to the underlying data model and is how other clients learn that it exists.
 
-### When to use one vs multiple objects
+Dynamically created objects need to be stored on an already connected shared object, so the most common case is to store them in an `initialObject`, because `initialObjects` are connected on creation. However, you can also store dynamic objects in other connected dynamic objects. In this sense shared objects are arbitrarily nestable.
 
-### Data objects are nestable
+When retrieving dynamically created objects you need to first get the object's handle then get the object from the handle. This reference based approach allows the Fluid Framework to virtualize the data underneath, only loading objects when they are requested.
+
+```typescript
+const schema = {
+    name: "example-container",
+    initialObjects: {
+        map: SharedMap,
+    },
+    dynamicObjectTypes: [ SharedCell ],
+}
+
+const { container, containerServices } = await client.getContainer(/*service config*/, schema);
+const map = container.initialObjects.map;
+
+const newCell = await container.create(SharedCell); // Create a new SharedCell
+map.set("cell-id", newCell.handle); // Attach the new SharedCell 
+
+// ...
+
+const cellHandle = map.get("cell-id"); // Get the handle
+const cell = await cellHandle.get(); // Resolve the handle to get the object
+
+// or
+
+const cell = await map.get("cell-id").get(); // Get and resolve handle
+
+// Listening for new dynamic objects
+
+map.on("valueChanged", (changed) => {
+    if (changed.key === "cell-id") {
+        const handle = map.get(changed.key);
+        handle.get().then((cell) => {
+            // ...
+        });
+    }
+}
+
+```
+
+### When to use dynamic objects
+ 
+Dynamic objects are more difficult to work with than `initialObjects`, but are especially important for large data sets where portions of the data are virtualized. Because dynamic objects are loaded into memory on demand, using them can reduce boot time of your application by delaying when the objects are loaded. Dynamic objects are also not strictly defined in the container schema. This enables you to create containers with flexible, user-generated schemas.
 
-## Versioning and upgrading schema
+An scenario where this is useful is if you were building a collaborative storyboarding application. In this scenario you can have a large number of boards that make up the larger storyboard. By using a dynamic shared objects for each board you can load them on demand as the user accesses them instead of having to load them all in memory at once.
```

Comment:
```suggestion
An example where this is useful is building a collaborative storyboarding application. In this scenario you can have a large number of individual boards that make up the storyboard. By using a dynamic shared objects for each board you can load them on demand as the user accesses them instead of having to load them all in memory at once.
```